### ISSUE TITLE
WinRM IIS Web Deploy - Only deploys empty App_Data folder

### ISSUE THREAD

**Issue by perp1exed** [Microsoft/vsts-rm-extensions#114]
Hi,
My deployment package is there in the correct folder.

The Overide Parameters is set with the IIS wev app name as is the Website Name property.

After deployment runs successfully, only the App_Data folder is present and it is empty.

Running on prem.

Please advise.
thanks,
KS

**Comment by mvvsubbu** [comment]
Can you Exclude files from App_Data option is selected which is present under MsDeploy Additional Options section

![image](https://cloud.githubusercontent.com/assets/11993108/19142717/39ab710a-8bbd-11e6-9f21-386b0e1bf3e6.png)

**Comment by mvvsubbu** [comment]
Can you update your state ?

**Comment by mvvsubbu** [comment]
Any update???

**Comment by perp1exed** [comment]
Sorry about the delay. Checking that option does not change the behaviour.

The App_Data folder is deployed with nothing in it.

thanks.

**Comment by pnguyen535** [comment]
Hi,

I ran into the issue below. Please help

 Deployment failed on machine  "machinexxxxx" with following message : System.Exception: Cannot bind argument to parameter 'Path' because it is null.

**Comment by perp1exed** [comment]
Could there be something wrong with the deployment zip file? It looks ok, has the standard directory structure. Maybe permissions issue? accept the account I use has rights to all servers. I'm confused.

**Comment by mvvsubbu** [comment]
Creating package you refer the following link for detailed help : https://www.visualstudio.com/en-us/docs/release/examples/other-servers/net-to-vm 

If you want to create package from visual studio you can follow : https://codemonkeygotsmarts.wordpress.com/2014/04/21/msdeploy-create-a-web-deployment-package-from-visual-studio/ 

If you have followed one of the above approach then there should not be any issue with package. Can you share your logs and package by setting system.debug to true

**Comment by mvvsubbu** [comment]
@pnguyen535,

When did you encounter the issue for Path is null exception?? Actually we have fixed similar issue on 13th October. If you have encountered the issue post that can you open a separate issue for that and share your deployment logs by setting system.debug to true.

**Comment by perp1exed** [comment]
I am creating the package via ALM release web portal - is that what you are asking?

The zip package is there in the stated location.

I have posted some settings - what other settings do you want to see?

thanks,

KS

**Comment by perp1exed** [comment]
Attaching log.
[ReleaseLogs_62.zip](https://github.com/Microsoft/vsts-rm-extensions/files/535258/ReleaseLogs_62.zip)

**Comment by perp1exed** [comment]
Just thinking out aloud here but my build outputs two folders: "_PublishedWebsites" and 'drop".

The contents of both of these are identical: There is a folder with site files and then there is a package folder containing the zip. 

What is the point of the folder with the website files only? It does not get deployed so why is it there? 

I ask because it would be easy to replace the tokens in the website files folder, if only it would be used for deployment.

Thanks,
KS
![image 1](https://cloud.githubusercontent.com/assets/22105451/19657204/27b3814e-99f1-11e6-838a-c5d79da5886c.png)
![image 2](https://cloud.githubusercontent.com/assets/22105451/19657203/27b2f29c-99f1-11e6-9ac0-e1565f904add.png)

**Comment by mvvsubbu** [comment]
First of all sorry for very late reply. I am not sure still you are facing any issue or not. I completely missed this issue. 

Using this extension you can now deploy folder as well, so try using the folder as input and let us know if you still face any issue.

**Comment by mvvsubbu** [comment]
Closing issue as I didn't heard, hope you got unblocked. Please feel free to reopen if the issue still persists
